http_handler.go

```diff
@@ -53,25 +81,44 @@ func (lvl AtomicLevel) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		enc.Encode(payload{Level: &current})
 
 	case http.MethodPut:
-		var req payload
-
-		if errmess := func() string {
-			if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
-				return fmt.Sprintf("Request body must be well-formed JSON: %v", err)
-			}
-			if req.Level == nil {
-				return "Must specify a logging level."
+		requestedLvl, err := func(body io.Reader) (*zapcore.Level, error) {
+			switch r.Header.Get("Content-Type") {
+			case "application/x-www-form-urlencoded":
+				pld, err := ioutil.ReadAll(body)
+				if err != nil {
+					return nil, err
+				}
+				values, err := url.ParseQuery(string(pld))
+				if err != nil {
+					return nil, err
+				}
+				lvl := values.Get("level")
+				if lvl == "" {
+					return nil, fmt.Errorf("must specify logging level")
+				}
+				var l zapcore.Level
+				if err := l.UnmarshalText([]byte(lvl)); err != nil {
+					return nil, err
+				}
+				return &l, nil
```

Comment:
Optionally, this can be shortened.
```suggestion
				err := l.UnmarshalText([]byte(lvl))
				return &l, err
```

Reply:
I prefer being explicit here. Especially after switching to value type.

It is not immediately obvious whether the returned value is pre- or post-UnmarshalText without knowing that part of the spec by hart. (TBH I would need to look that up)